src/test/java/hudson/plugins/git/extensions/impl/WipeWorkspaceTest.java

```diff
@@ -1,9 +1,52 @@
 package hudson.plugins.git.extensions.impl;
 
+import hudson.EnvVars;
+import hudson.model.FreeStyleBuild;
+import hudson.model.FreeStyleProject;
+import hudson.model.Result;
+import hudson.plugins.git.GitSCM;
+import hudson.plugins.git.TestGitRepo;
+import hudson.plugins.git.extensions.GitSCMExtension;
+import hudson.plugins.git.extensions.GitSCMExtensionTest;
 import nl.jqno.equalsverifier.EqualsVerifier;
+import org.jenkinsci.plugins.gitclient.Git;
+import org.jenkinsci.plugins.gitclient.GitClient;
 import org.junit.Test;
 
-public class WipeWorkspaceTest {
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+
+public class WipeWorkspaceTest extends GitSCMExtensionTest {
+
+    TestGitRepo repo;
+    GitClient git;
+
+    @Override
+    public void before() throws Exception {
+        repo = new TestGitRepo("repo", tmp.newFolder(), listener);
+        git = Git.with(listener, new EnvVars()).in(repo.gitDir).getClient();
+    }
+
+    @Override
+    protected GitSCMExtension getExtension() {
+        return new WipeWorkspace();
+    }
+
+    /**
+     * Test to confirm the behavior of forcing re-clone before checkout by cleaning the workspace first.
+     **/
+    @Test
+    public void testWipeWorkspace() throws Exception {
+        FreeStyleProject projectWithMaster = setupBasicProject(repo);
+        WipeWorkspace wipeWorkspaceExtension = new WipeWorkspace();
+        ((GitSCM)projectWithMaster.getScm()).getExtensions().add(wipeWorkspaceExtension);
+
+        git.commit("First commit");
+        FreeStyleBuild build = build(projectWithMaster, Result.SUCCESS);
+
+        String buildLog = build.getLog();
+        assertThat("Workspace not cleaned before checkout",true, is(buildLog.contains("Wiping out workspace first.")));
+    }
 
     @Test
```

Comment:
The change of class inheritance switches this test from a fast test to a slow test.  That happens because the test now uses a JenkinsRule that it did not use previously. The equalsContract test does not need a JenkinsRule, so there is an annotation that should be used to avoid the cost of JenkinsRule for a test that does not need it.  It will need an import elsewhere in the file.
```suggestion
    @Test
    @WithoutJenkinsRule
```

Reply:
Change looks good.  Thanks!  I'll plan to merge it next week when I return from my trip.  Won't merge until at least Monday so that I have time to check the behavior in my environment.